apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -43,28 +44,53 @@ pub(crate) struct BridgeQueryPlanner {
 
 impl BridgeQueryPlanner {
     pub(crate) async fn new(
-        schema: String,
+        sdl: String,
         configuration: Arc<Configuration>,
     ) -> Result<Self, ServiceBuildError> {
-        let planner = Arc::new(
-            Planner::new(
-                schema.clone(),
-                QueryPlannerConfig {
-                    incremental_delivery: Some(IncrementalDeliverySupport {
-                        enable_defer: Some(configuration.supergraph.defer_support),
-                    }),
-                },
-            )
-            .await?,
-        );
+        let schema = Schema::parse(&sdl, &configuration)?;
+
+        let planner = Planner::new(
+            sdl,
+            QueryPlannerConfig {
+                incremental_delivery: Some(IncrementalDeliverySupport {
+                    enable_defer: Some(configuration.supergraph.defer_support),
+                }),
+                graphql_validation: matches!(
+                    configuration.experimental_graphql_validation,
+                    GraphQLValidation::Legacy | GraphQLValidation::Both
+                ),
+            },
+        )
+        .await;
+
+        let planner = match planner {
+            Ok(planner) => planner,
+            Err(err) => {
+                if configuration.experimental_graphql_validation == GraphQLValidation::Both {
+                    let has_validation_errors = err.iter().any(|err| err.is_validation_error());
+
+                    assert!(
```

Comment:
don't make it an assert. We should not crash the router if an invalid schema was received

Reply:
Yeah i will add the log message + metric here instead